### DISCUSSION TITLE
How to clear queryClient outside component

### DISCUSSION THREAD

**Discussion by snax4a** [TanStack/query#3551] [Q&A]
Hello,

I would like to clear queryClient cache outside rect component.
In component i am doing the following:
```js
const queryClient = useQueryClient()
queryClient.clear()
```
I would like to clear it in my Axios response interceptor when I get the 401 status code but I can't use react hook there.
Do you have any suggestions on how I can achieve that?

**Comment by TkDodo** [comment]
My best guess is that the axios instance and the interceptors should be created inside a `useAxios` hook so that you can `useQueryClient` in that hook.
